docs/user/templates/webhook/01_webhook_config.md

```diff
@@ -0,0 +1,75 @@
+<!-- page-title: Webhook -->
+
+# Webhook
+
+When an event happens (e.g. creating new model's version, deploy endpoint), we might want to trigger action in other services, Merlin provides a way this by triggering a webhook call. 
+The webhook that will be called is configured per event type, and it could contain multiple endpoints which can be configured as synchronous or asynchronous call. 
+
+For the detail of available configuration please refer to [MLP library docs](https://github.com/caraml-dev/mlp/blob/main/api/pkg/webhooks/README.md), this documentation will only cover the basic config and the paylod of webhook which we has implemented.
+
+## 1. Webhook Configuration
+In Merlin's yaml for config you can add new field for the webhook's configuration which will looks like this:
+```yaml
+WebhooksConfig:
+  Enabled: true
+  Config:
+    on-model-version-deployed:
+      - URL: http://127.0.0.1:8000/async-webhook
+        Method: POST
+        Async: true
+        Name: async
+    on-model-version-undeployed:
+      - URL: http://127.0.0.1:8000/sync-webhook
+        Method: POST
+        FinalResponse: true
+        Name: sync
+```
+## 2. Webhook Events and Payload
+### 2.1. Model Version's Endpoint
+**Events:**
+- `on-model-version-predeployment`: will be triggered before the version endpoint is deployed, if the sync webhook return an error (non `200` response), the deployment will be **aborted**. The successful response of this webhook will not affect the deployment process, as we do not process the response
+- `on-model-version-deployed`: will be triggered after the version endpoint is successfully deployed (either new or updated)
+- `on-model-version-undeployed`: will be triggered after the version endpoint is successfully undeployed
```

Comment:
Tiny nit on this - could we first specify what is expected to happen normally (i.e. 200 response) before explaining what would happen in a scenario where a webhook returns an error? 😅 For example,

- `on-model-version-predeployment` - will be triggered before the version endpoint is deployed, and requires a 200 response from the webhook for the deployment to proceed. If any of the sync webhooks return an error (non `200` response), the deployment will be aborted. In both cases the responses of the webhooks will not be processed.

We can do the same for the other 2 descriptions for `on-model-version-deployed` and `on-model-version-undeployed` too (they don't have information on what the outcome would be if the webhooks return a success or error).